migration_original/ODS1Stage/tables/Show/SOLRGeographicArea/spu_translated_SOLRGeographicArea.sql

```diff
@@ -0,0 +1,128 @@
+-- spuSOLRGeographicAreaGenerateFromMid
+CREATE OR REPLACE PROCEDURE SHOW.SP_LOAD_SOLRGEOGRAPHICAREA() 
+    RETURNS STRING
+    LANGUAGE SQL
+    AS  
+
+DECLARE 
+
+---------------------------------------------------------
+--------------- 0. Table dependencies -------------------
+---------------------------------------------------------
+    
+-- Show.SOLRGeographicArea depends on: 
+--- Show.SOLRGeographicAreaDelta
+--- Mid.GeographicArea
+
+---------------------------------------------------------
+--------------- 1. Declaring variables ------------------
+---------------------------------------------------------
+
+    select_statement STRING; -- CTE statement
+    update_statement STRING;
+    insert_statement STRING;
+    merge_statement STRING; -- Insert statement to final table
+    status STRING; -- Status monitoring
+
+   
+---------------------------------------------------------
+--------------- 2.Conditionals if any -------------------
+---------------------------------------------------------   
+   
+BEGIN
+    -- no conditionals
+
+
+---------------------------------------------------------
+----------------- 3. SQL statements ---------------------
+---------------------------------------------------------     
+
+-- Select Statement
+select_statement := 'WITH CTE_geoId AS (SELECT
+    DISTINCT GeographicAreaID
+FROM
+    Show.SOLRGeographicAreaDelta
+WHERE
+    StartDeltaProcessDate IS NULL
+    AND EndDeltaProcessDate IS NULL
+    AND SolrDeltaTypeCode = 1 --INSERT/UPDATEs
+    AND MidDeltaProcessComplete = 1 ) --this will indicate the Mid TABLEs have been refreshed with the updated data
+
+SELECT
+            midGeo.GeographicAreaID,
+            midGeo.GeographicAreaCode,
+            midGeo.GeographicAreaTypeCode,
+            midGeo.GeographicAreaValue,
+            CURRENT_TIMESTAMP AS UpdatedDate,
+            CURRENT_USER AS UpdatedSource
+        FROM
+            Mid.GeographicArea midGeo
+            JOIN (SELECT GeographicAreaID FROM CTE_geoId) as geoId on geoId.GeographicAreaID = midGeo.GeographicAreaID';
```

Comment:
I think this could be a simple join, no need of the subquery. Also use uppercase for the ON statement
```suggestion
            JOIN CTE_geoId as geoId ON geoId.GeographicAreaID = midGeo.GeographicAreaID';
```

Reply:
Done